gitpod-io/gitpod
# [proxy] add analytics endpoint

## Description


[Internal context](https://gitpod.slack.com/archives/C047WL9JBLG/p1680019749050709)

- This PR introduces a new `/analytics` endpoint for collecting analytics from different clients in a centralized manner. The endpoint utilizes basic authentication from segment to determine whether to proxy to trusted or untrusted sources.
- In preview environments, Gitpod components (server, ws-manager-bridge, ws-proxy) are configured to use this endpoint with a trusted key, and the same configuration will be applied to production environments in a subsequent PR.
- In workspace services (supervisor and Gitpod CLI) are also reconfigured to use this endpoint but without a key in preview environments (i.e., untrusted), and the same configuration to be applied to the production environment in a subsequent PR.
- After deployment, local clients like VS Code Desktop will also be updated to use this endpoint.
- This PR as well fixes analytics installer options for GH actions.

## Related Issue(s)

Fixes #

## How to test


- Start a new workspace and check that analytics is configured now based on segment endpoint not key.
<img width="1060" alt="Screenshot 2023-04-12 at 21 39 29" src="https://user-images.githubusercontent.com/3082655/231568577-7095bfa6-0c35-4e11-8a1d-ad89cf5cf20b.png">

- Run gp tasks list command and then check staging trusted that gp_command event is registered there.
- Check that events from other Gitpod components are registered as well, i.e. workspace started and stopped, and so on.

## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
